CFC_WebApp/main/mode/cluster_pipeline.py

```diff
@@ -0,0 +1,327 @@
+    y2 = np.mean(y_coords) + 10*np.std(y_coords)
+
+    plt.axis((x1, x2, y1, y2))
+
+    if is_ground_truth:
+        f_name = 'mds_plots/ground_truth_' + str(user_id) + '_' + method + '.png'
+    else:
+        f_name = 'mds_plots/' + str(user_id) + '_' + method + '.png'
+    
+    plt.savefig(f_name)
+
+#K MEANS HELPER FUNCTIONS
+
+def generate_section_matrix(user_id):
+    sections = get_user_sections(user_id)
+    inv_feature_dict = {0: 'start_lat', 1: 'start_lng', 2: 'end_lat', 3: 'end_lng', 4: 'duration', 5: 'distance'}
+    feature_dict = {'start_lat': 0, 'start_lng': 1, 'end_lat': 2, 'end_lng': 3, 'duration': 4, 'distance': 5}
+
+    data = np.zeros((len(sections), len(feature_dict)))
+
+    c = 0
+    while c < len(sections):
+        section = sections[c]
+
+        start_point = section['track_points'][0]['track_location']['coordinates']
+        end_point = section['track_points'][-1]['track_location']['coordinates']
+        start_lat = start_point[1]
+        start_lng = start_point[0]
+        end_lat = end_point[1]
+        end_lng = end_point[0]
+        duration = section['duration']
+        distance = section['distance']
+        data[c][feature_dict['start_lat']] = start_lat
+        data[c][feature_dict['start_lng']] = start_lng
+        data[c][feature_dict['end_lat']] = end_lat
+        data[c][feature_dict['end_lng']] = end_lng
+        data[c][feature_dict['duration']] = duration
+        data[c][feature_dict['distance']] = distance    
+        c += 1
+    return data
+
+def kmeans(data):
+    sections = get_user_sections(user_id)
+    k_means = KMeans(init='k-means++', n_clusters=3, n_init=10)
+    k_means.fit(data)
+    k_means_labels = k_means.labels_
+    c = 0
+    clusters = {}
+    while c < k_means_labels.shape[0]:
+        if k_means_labels[c] not in clusters:
+            clusters[k_means_labels[c]] = [sections[c]]
+        else:
+            clusters[k_means_labels[c]].append(sections[c])
+        c += 1
+    return clusters
+
+def user_route_data2(section_ids):
+    data_feature = {}
+
+    # for section in database.find({'$and':[{'user_id': user_id},{'type': 'move'},{'confirmed_mode': {'$ne': ''}}]}):
+    for _id in section_ids:
+        try:
+            data_feature[_id] = getRoute(_id)
+        except Exception as e:
+            pass
+    #print(data_feature.keys())
+    return data_feature
+
+#########################################################################################################
+                                # END OF LOW LEVEL ABSTRACTION #
+#########################################################################################################
+
+
+#user_list = get_user_list()
+
+user_uuid = read_uuids()
+if len(sys.argv) == 2:
+    user_id = user_uuid[sys.argv[1]]
+
+print(user_id)
+
+#PARAMETERS
+methods = ['dtw', 'lcs', 'Frechet'] #what metrics for distance to use
+
+#EXPERIMENT 1: KMeans with following features: start_lat, start_lng, end_lat, end_lng, duration, distance
```

Comment:
why use different clustering techniques for featurized and trajectory matching metrics?